test/Groups-conformance-tests.jl

```diff
@@ -0,0 +1,298 @@
+function test_Group_interface(G::Group)
+    @testset "Group interface" begin
+#        @testset "Iteration protocol" begin
```

Comment:
I disabled these iteration tests for now as we don't implement all of that generically right now, and I am not sure we really want it that way.